npm/ng-packs/packages/oauth/src/lib/guards/oauth.guard.ts

```diff
@@ -53,3 +53,33 @@ export const abpOAuthGuard: CanActivateFn = (
   authService.navigateToLogin(params);
   return false;
 };
+
+export const asyncAbpOAuthGuard: CanActivateFn = (
+  route: ActivatedRouteSnapshot,
+  state: RouterStateSnapshot,
+) => {
+  const authService = inject(AuthService);
+  const code = new URLSearchParams(window.location.search).get('code');
+
+  if (code) {
+    return firstValueFrom(
+      interval(100).pipe(
+        map(() => authService.isAuthenticated),
```

Comment:
We should still use `oAuthService.hasValidAccessToken()` here where we can simply use `filter(Boolean)` after that to keep it synced with the other condition and the guard.